### PR TITLE
Add externalIpIsAvailable to status DTO

### LINKED ISSUE #662: Fix Public IP bug for undeployed new application (version)
Currently the application tab shows public IPs for a new application version, if it is undeployed.

### LINKED ISSUE #664: Improve response DTO for GET /publicIP
Endpoint `GET /services/{{serviceShortName}}/{{serviceVersion}}/interfaces/{{serviceInterfaceName}}/publicIP` MUST NOT return a 4xx status code if there is no public IP available for the interface. Instead it should response with a 2xx status code with a DTO that includes an information whether the public IP is currently in `pending` state.

Current response:
```
{
    "timestamp": "2019-03-26T16:21:28.662+0000",
    "status": 404,
    "error": "Not Found",
    "message": "No deployed service interface 'http' of MicoService 'react-redux-realworld-example-app' 'v1.0.0' was found!",
    "path": "/services/react-redux-realworld-example-app/v1.0.0/interfaces/http/publicIP/"
}
```

### PR DESCRIPTION
**Pull request by marvin21**
- [X] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [X] Ensure to use auto format in **all** files
- [X] Tests created for changes or:
  - [X] Test cases adapted
---

## Short Description

To improve the status response for MicoServiceInterfaces during the allocation of the public IP address in Kubernetes, an additional field `externalIpIsAvailable` is added to `MicoServiceInterfaceStatusResponseDTO`. If the boolean is set to true, the external IP is available in Kubernetes. If the value of this boolean is false, the IP address is in a pending state. No error message is returned anymore, but an adapted DTO with information about the state of the IP instead.

## Resolving Issue/ Feature

Resolves #664 
Resolves #662

### CODE REVIEW

**Comment by marvin21** [comment]
@gepperho : 
We have now an additional field `externalIpIsAvailaible` in `MicoServiceInterfaceStatusResponseDTO`, which is set to `true `if the IP address can be retreived from Kubernetes. Otherwise, this field is set to `false` per default. 
When the IP is in the pending state, the endpoint does not return an exception anymore, but a DTO, where the field `externalIp` is null and  `externalIpIsAvailable`  is `false`.

**Comment by gepperho** [comment]
when `externalIpIsAvailable` is `false`, will the ip be available at a later point in time? Or more precisely: could a service interface exist without a publicIp?

**Comment by marvin21** [comment]
No, eventually, all service interfaces have a public IP. In the context of MICO we do not have Kubernetes services without public IPs.

**Comment by davidkopp** [comment]
> when `externalIpIsAvailable` is `false`, will the ip be available at a later point in time? Or more precisely: could a service interface exist without a publicIp?

The answer is yes. If `externalIpIsAvailable` is `false` the IP is in pending state and will be available some seconds later. So a MICO service interface can be exist without a public IP, but only for some seconds (maximal 1-2 minutes).

File: mico-admin/src/app/app-detail-public-ip/app-detail-public-ip.component.ts [hunk 1]
```diff
@@ -77,11 +77,13 @@ export class AppDetailPublicIpComponent implements OnInit, OnChanges, OnDestroy
                     application.services.forEach(service => {
 
                         safeUnsubscribeList(this.subServiceInterfaces);
+                        this.subServiceInterfaces = [];
                         // assumption: one public ip per interface
                         this.subServiceInterfaces.push(this.apiService.getServiceInterfaces(service.shortName, service.version)
                             .subscribe(serviceInterfaces => {
 
                                 safeUnsubscribeList(this.subPublicIps);
+                                this.subPublicIps = [];
```
Comment (buehlefs) [inline mico-admin/src/app/app-detail-public-ip/app-detail-public-ip.component.ts]: move (together with safeUnsubscribe list) out ouf subscription to capture all ip subscriptions in the same list.

**Pull request merged**